### PR TITLE
Fix BEM class naming

### PR DESCRIPTION
**Pull request by meowsus**
Class names like `1/2`, `50` and `1-of-2` pollute the global namespace.
These classes describe cell widths, so make them be modifiers of the
`grid__cell` element:
- grid__cell--1/2
- grid__cell--50
- grid__cell--1-of-2

Modifiers should _modify_ a block or an element. They should not be used
by themselves. Consider the markup:

```
<div class="grid">
  <div class="1/2 1/4--wide grid__cell">Cell</div>
  <div class="1/2 1/4--wide grid__cell">Cell</div>
  <div class="1/2 1/4--wide grid__cell">Cell</div>
  <div class="1/2 1/4--wide grid__cell">Cell</div>
</div>
```

The `wide` modifier is being used on a `1/4` block, but the block itself
is not referenced. Instead it's understood that the `grid__cell` is
`1/2` until `wide` when it becomes `1/4`. The classes used in the above
example are disassociative.

Improve this concept:

```
<div class="grid">
  <div class="grid__cell grid__cell--1/2 grid__cell--1/4-for-wide">Cell</div>
  <div class="grid__cell grid__cell--1/2 grid__cell--1/4-for-wide">Cell</div>
  <div class="grid__cell grid__cell--1/2 grid__cell--1/4-for-wide">Cell</div>
  <div class="grid__cell grid__cell--1/2 grid__cell--1/4-for-wide">Cell</div>
</div>
```

This example more accurately expresses the width + responsive logic,
using BEM-like naming.

### CODE REVIEW

**Comment by meowsus** [comment]
This PR doesn't include a version of `avalanche.min.css` because it's unclear what you are using to minify your code. If you'd like I can add a gulpfile and include a minification task with this (or another) PR.

**Comment by colourgarden** [comment]
Hi @meowsus. Thanks for this thought-provoking input.

One of the primary motivating factors behind Avalanche was my dissatisfaction with the verbosity of the width class naming of other grids I'd used. Nesting `col-8` inside `col-9` and ending up with a 50% total width made no sense to me. So I set out to make something that was easily human readable and I think `1/2 grid__cell` achieves that.

You are correct when you say width classes aren't blocks but bear in mind that the widths are a utility, not just a modifier. They can be used independently of the grid should the developer wish. There's probably an argument about the grid system not overstepping its boundaries but I've found it useful to have responsive width classes available in other situations outside the grid.

Your point about the global namespace is certainly valid. Avalanche provisions the [`$av-width-class-namespace`](https://github.com/colourgarden/avalanche/blob/master/_avalanche.scss#L10) variable which will prepend your chosen namespace to all width classes.

Were you to set this variable to `#{$av-namespace}__cell--`, your desired markup could be largely achieved, as the following example class would be produced:

`.grid__cell--1/4--thumb`

However, this introduces double modifier classes which aren't really desirable.

In summary, I don't wish to compromise the easily read nature of Avalanche just to more tightly adhere to the principles of BEM.

Thanks very much for your support of Avalanche. Please do let me know if you have any live projects out there using the library.

Ps. I do use Gulp myself for building Avalanche but I don't wish to prescribe a specific tool to people. If they're using Sass then they've likely already got a build process/app of some sort in place which Avalanche can be dropped into.

**Comment by meowsus** [comment]
This all makes sense. Your approach is very well thought out.

The only reason I mentioned Gulp was for contributors, so that they could build and commit the updated, minified CSS file along with their work.

You have a great grid system here. I may end up hard forking this at some point. If I do, I'd love your input.

**Comment by meowsus** [comment]
Here's the hard fork: http://github.com/meowsus/grip

**Comment by colourgarden** [comment]
Do let me know if you need any help with anything or if you implement any cool features you think should be included! Thanks for supporting the project.

**Comment by meowsus** [comment]
Of course! And likewise, of course.
